src/Plugin/EntityBrowser/Widget/Webdam.php

```diff
@@ -0,0 +1,286 @@
+        $assets[$folder_item->id] = '<p class="webdam-asset-fielname">'.$folder_item->name.'</p>';
+        //If the asset has a thumbnail it should be rendered
+        if(!empty($folder_item->thumbnailurls)){
+          //Multiple thumbnail sizes are available.  Using the 220 width version for now
+          $assets[$folder_item->id] .= '<img class="webdam-asset-thumbnail" src="'.$folder_item->thumbnailurls[2]->url.'" alt="'.$folder_item->name.'" />';
+        }
+      }
+    }
+    //If the assets array is not empty then add the assets to the form as checkboxes
+    if(!empty($assets)){
+      $form['asset-container']['assets'] = [
+        '#type' => 'checkboxes',
+        '#title' => $this->t('Choose one or more assets'),
+        '#title_display' => 'invisible',
+        '#options' => $assets,
+        // Multiple assets will only be accepted if the source field allows more than one value.
+        '#multiple' => $field_cardinality != 1 && $this->configuration['multiple'],
+      ];
+    }
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function prepareEntities(array $form, FormStateInterface $form_state) {
+    foreach ($form_state->getValue(['upload'], []) as $aid) {
+      if ($aid !== 0) {
+        //TODO: validate form selection
+      }
+    }
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submit(array &$element, array &$form, FormStateInterface $form_state) {
+    $assets = [];
+    if (!empty($form_state->getTriggeringElement()['#eb_widget_main_submit'])) {
+      foreach ($form_state->getValue(['assets'], []) as $aid) {
+        if ($aid !== 0) {
+          $webdam_asset = $this->webdam->getAsset($aid);
+          ksm($webdam_asset);
+          $media_asset = Media::create([
+            'bundle' => 'webdam',
+            'uid' => '1',
+            'langcode' => 'en',
+            'status' => Media::PUBLISHED,
+            'name' => $webdam_asset->name,
+            'field_asset_id' => $aid,
+          ]);
+          $media_asset->save();
+          $assets[] = $media_asset;
+        }
+      }
+    }
+    // $this->clearFormValues($element, $form_state);
+    $this->selectEntities($assets, $form_state);
+  }
+
+  /**
+   * Clear values from upload form element.
+   *
+   * @param array $element
+   *   Upload form element.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state object.
+   *
+   * TODO: Determine if this function is still needed.  It was initially copied from Upload widget class in the entity browser module
+   *
+   */
+  protected function clearFormValues(array &$element, FormStateInterface $form_state) {
+    // We propagated entities to the other parts of the system. We can now remove
+    // them from our values.
+    // $form_state->setValueForElement($element['upload']['fids'], '');
+    // NestedArray::setValue($form_state->getUserInput(), $element['upload']['fids']['#parents'], '');
+  }
```

Comment:
I think we can nix this entire function. It doesn't seem to be required by anything.

Reply:
Removed